### PR TITLE
chore!: reduce bundle size

### PR DESCRIPTION
**Pull request by matthieuh**
## Description

Done in this PR:
- Give the ability to do tree shaking using this lib: 
    => usage of `@rollup/plugin-babel` to use `@babel/plugin-transform-react-pure-annotations` (cf. https://github.com/babel/babel/pull/11428). Without this  usage of `React.forwardRef` or `React.memo` is breaking tree-shaking.
- Fix to have tree shaking working on server-side of Next.js and due to this [issue](https://github.com/vercel/next.js/issues/12557) (tree shaking broken when lib is using barrel files)
    => avoid 1 file bundle and build the lib in separate files (`preserveModules: true`)
- Remove bundled fonts (cf. https://github.com/radix-ui/design-system/issues/148)

## Dependency changes

| Dependency | Version        | State   |
| ---------- | -------------- | ------- |
| @babel/plugin-transform-react-pure-annotations       | ^7.16.7         | Added   |
|@rollup/plugin-babel     | ^5.3.1       | Added   |
| rollup       | ^2.33.1 -> ^2.70.1 | Updated |
| rollup-plugin-url       |     | Removed |

## Breaking changes

- Fonts have to be imported into your main project
## How to test?

1. On a basic `Next.js` project add `config.resolve.alias.react = path.resolve(__dirname, 'node_modules/react')` in the custom webpack config. Use some components from Faency both in `_app.ts` and in some nextjs page (and be sure there are some you are not using).
2. In the `Faency` repo run `npm run build`
3. In the `Faency` repo run `yarn link`
4. In the `Next.js` repo run `yarn link @traefiklabs/faency`
5. In the `Next.js` repo run `yarn build`
6. Analyze `.next` folder and see if faency components that you are not using are not present. `@next/bundle-analyzer` can be a good way to analyze it.

## Good PR checkboxes

- [x] Change has been tested
- [ ] Added/Updated tests
- [ ] Added/Updated stories
- [x] PR follows [conventions](https://github.com/traefik/faency#how-to-contribute)
- [x] Labels are set
- [x] Project is linked

## Good Review checkboxes

<details>
<summary> ℹ️  Copy the snippet and paste in the review field to fill it</summary>

```markdown
- [ ] I've tested the changes
- [ ] I've agreed on the unit tests (soon to come)
- [ ] I've checked the stories
- [ ] I've read the code and understood it
- [ ] I don't have any more questions
- [ ] I've described any optional improvements
- [ ] I checked PR follows [conventions](https://github.com/traefik/faency#how-to-contribute)
```

</details>

### CODE REVIEW

File: components/AriaTable/AriaTable.stories.tsx [hunk 1]
```diff
@@ -527,32 +524,3 @@ export const Columns: ComponentStory<any> = ({ transform, ...args }) => (
...
   </Flex>
 );
-
-const FlexIssue = () => (
-  <Flex align="center" justify="center">
-    <Image src="https://picsum.photos/38/38" />
-  </Flex>
-);
-
-export const VerticalAlignment: ComponentStory<any> = (args) => (
```
Comment (paulocfjunior) [inline components/AriaTable/AriaTable.stories.tsx:537]: Why did we remove this story?

File: components/Table/Table.tsx [hunk 2]
```diff
@@ -19,7 +19,6 @@ export const Tbody = styled('tbody', {
 export const Th = styled('th', Label, {
   // override Label
   display: 'table-cell',
-  verticalAlign: 'inherit',
```
Comment (paulocfjunior) [inline components/Table/Table.tsx:22]: This seems to be reverting this PR:
- https://github.com/traefik/faency/pull/331

Is that intentional?

File: components/AriaTable/AriaTable.stories.tsx (hunk 1 above)
Comment (matthieuh) [reply components/AriaTable/AriaTable.stories.tsx]: It was totally unintentional :P

File: components/Table/Table.tsx (hunk 2 above)
Comment (matthieuh) [reply components/Table/Table.tsx]: It was totally unintentional :P

**Review by paulocfjunior — APPROVED** [review]
Before:
Screen Shot 2022-04-11 at 01 03 24

After:
Screen Shot 2022-04-11 at 01 02 08

**Comment by paulocfjunior** [comment]
@matthieuh I think we can also get rid of the `types` folder, as it's currently only used to define namespaces for the fonts.

**Review by seedy — APPROVED** [review]
- Before
image

- After
image

**Pull request merged**